### PR TITLE
Add tests for base58

### PR DESCRIPTION
**Pull request by hodlwave**
Hi Stepan,

I thought I'd start adding some test vectors for your bitcoin library. I added the project's root directory to the path in a very hacky way (see `tests/run_tests.py`) because there weren't any pre-existing examples and I wasn't sure how to do it. I'm guessing there's better way to do that...

### CODE REVIEW

File: tests/run_tests.py [hunk 1]
```diff
@@ -1,3 +1,8 @@
+import sys
 import unittest
 
+curdir = sys.path[0]
+pardir = curdir + '/..'
+sys.path.append(pardir)
```
Comment (stepansnigirev) [inline tests/run_tests.py:6]: Maybe let's keep it the same as I do in specter-diy,
then all imports will be the same as in normal firmware.

```suggestion
sys.path.append('../libs/common')
sys.path.append('../libs/unix')
sys.path.append('../usermods/udisplay_f469/display_unixport')
```

File: tests/tests/test_base58.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Tests adapted from base58:
+# https://github.com/keis/base58/blob/master/test_base58.py
+
+from libs.common.bitcoin.base58 import (
+    encode, decode, encode_check, decode_check)
```
Comment (stepansnigirev) [inline tests/tests/test_base58.py:5]: If we tweak `sys.path` like in my suggestion we can change it to
```suggestion
from bitcoin.base58 import encode, decode, encode_check, decode_check
```

**Review by stepansnigirev — COMMENTED** [review]
Looks great, thank you!
We definitely have not enough tests, so it's very needed indeed!
I left a small comment how I think we could add everything to sys.path, otherwise will be happy to merge!

File: tests/tests/test_base58.py (hunk 2 above)
Comment (hodlwave) [reply tests/tests/test_base58.py]: Fixed

File: tests/run_tests.py (hunk 1 above)
Comment (hodlwave) [reply tests/run_tests.py]: Fixed

**Comment by hodlwave** [comment]
Sounds good, updated the code as per your comments.

**Pull request merged**